src/browser/api_protocol/api_handlers/system.js

```diff
@@ -17,6 +17,35 @@ let apiProtocolBase = require('./api_protocol_base.js');
 let System = require('../../api/system.js').System;
 let _ = require('underscore');
 const connectionManager = require('../../connection_manager');
+const log = require('../../log');
+
+const ReadRegistryValuePolicyDelegate = {
+    //checkPermissions(ApiPolicyDelegateArgs): boolean;
+    checkPermissions: function(args) {
+        // permissionSettings has following format
+        // { "enabled": true, "registryKeys": [ "HKEY_CURRENT_USER\\Software\\OpenFin\\RVM" ] }
+        let permitted = false; // default to false
+        if (args.payload && args.permissionSettings && args.permissionSettings.enabled === true) {
+            if (Array.isArray(args.permissionSettings.registryKeys)) {
+                let fullPath = args.payload.rootKey;
+                if (args.payload.subkey) {
+                    fullPath = fullPath.concat('\\' + args.payload.subkey);
```

Comment:
Can you use `path.join(...` here?

Reply:
`path.join(...)` is not suitable for joining registry key paths since registry key names can contain characters that are invalid for directory names.

For example, if a registry key name contains a forward slash, `path.join` will replace it with a backslash, and the key won't be correctly accessed.